File: 2022/phase2_dress_rehearsal/byuvrx/src/frostlab-inekf-54148f8ac0c7/python/tests/Inertial/test_InertialProcess.py
import numpy as np
from numpy.testing import assert_allclose
from inekf import SE3, SO3, DVLSensor, InertialProcess
import pytest

def test_f():
    state = SE3[2,6]()
    u = np.array([1,1,1,1,1,1+9.81])

    ip = InertialProcess()
    result = ip.f(u, 1, state)

    assert_allclose(result.R().State, SO3.Exp(u[:3]).State)
    assert_allclose(result[0], np.ones(3))
    assert_allclose(result[1], np.ones(3)/2)
    

File: 2022/phase2_dress_rehearsal/byuvrx/src/frostlab-inekf-54148f8ac0c7/tests/Inertial/test_InertialProcess.cpp
#include "gtest/gtest.h"
#include <Eigen/Core>
#include <InEKF/Core>
#include <InEKF/Inertial>

#define EXPECT_MATRICES_EQ(M_actual, M_expected) \
  EXPECT_TRUE(M_actual.isApprox(M_expected, 1e-6)) << "  Actual:\n" << M_actual << "\nExpected:\n" << M_expected


TEST(InertialProcess, f){
    InEKF::SE3<2,6> state;
    Eigen::Matrix<double,6,1> u;
    u << 1,1,1,1,1,1+9.81;

    InEKF::InertialProcess ip;
    InEKF::SE3<2,6> result = ip.f(u, 1, state);

    EXPECT_MATRICES_EQ(result.R()(), InEKF::SO3<>::Exp(u.head(3))());
    EXPECT_MATRICES_EQ(result[0], Eigen::Vector3d::Ones());
    EXPECT_MATRICES_EQ(result[1], Eigen::Vector3d::Ones()/2);
}

// TODO: Find a good way to test makePhi?